OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -500,21 +518,34 @@ public void Dispose()
 		}
 	}
 
-	sealed class ThreadedVertexBuffer : IVertexBuffer<Vertex>
+	sealed class ThreadedVertexBuffer<T> : IVertexBuffer<T> where T : struct
 	{
+		readonly Stack<T[]> verticesPool = new();
```

Comment:
Moving this pool inside each buffer reduces it's utility - buffers with the same vertex type can't share from each other's pool.

Consider a `Dictionary<Type, Stack<object>>` in the graphics context, or perhaps some holder type `class VertexPool<T>` with a `Stack<T[]>` so we can maximise sharing.

Reply:
I think we definitely want to avoid boxing here. As for the holder class won't it require boxing as well?

Reply:
An array is a reference type, so putting it into an object doesn't require boxing.

Reply:
ah right, the original implementation I copied had casted to object[]

Reply:
You're quite right in that we don't want to use `object[]` for the vertex arrays. We don't want to have to cast every individual vertex back and forth. But stuffing different `T[]` types into `object` is fine we can cast the entire array to the correct underlying type when required and the elements themselves never have to be individually boxed/unboxed.

Reply:
managed to do it with `Dictionary<Type, object> vertexBufferPools`